### ISSUE TITLE
Expose request headers to template functions

### ISSUE THREAD

**Issue by rixo** [webpro/dyson#44]
Currently template functions receive only params, query, body, and cookies. That would be great to also expose request headers to them. I've got an example use case where I use dyson to back my client e2e tests, and I need to select a response based on Authorization header.

In my fork, I've just [added a fifth parameters for the headers](https://github.com/webpro/dyson/compare/master...rixo:expose-headers-to-templates), and it works. Unfortunately, that makes a lot of (potentially unused) arguments, which is not so great... In my opinion, it would be cleaner to expose an object wrapping these 5 arguments, or even the request object itself... even though I see how that could be considered hazardous. Unfortunately, those 2 solutions would be breaking for existing code bases. Maybe as an option? From what I've found, all calls are made in the same file, so that would probably be doable.

Let me know what you think of the idea, I can work a PR if needed.

**Comment by webpro** [comment]
Hi @rixo, can't believe I've ignored this friendly and helpful input. Sincere apologies.

Are you still interested in working the PR?

I'm all for it! Maybe a good option is to simply add an extra argument first, so it's a non-breaking change (v0.6.x). Then I'm fine with a breaking change (v1.0.0) and pass the object, just as you propose. Perhaps the request object itself could be passed with that object as well.

**Comment by rixo** [comment]
Hey! No worries.

I've already worked on this a bit. The approach I've eventually taken is slightly different from what you propose. I indeed added an option `exposeRequest`, that can be global or overridden in each service configuration ; if we let the default be false, then it stays non breaking.

Also, I do think that exposing the actual request object is more perennial than trying to wrap it. The latter solution would need future maintenance and I don't think JS developers generally need to be protected against themselves.

I'll send a PR with that soon. I still need to clean some bits and add some tests.
